src/apiserver/api_service_impl.h

```diff
@@ -0,0 +1,269 @@
+    Column() = default;
+    Column(std::string n, hybridse::sdk::DataType t, bool c, bool null)
+        : name(std::move(n)), type(t), is_constant(c), is_null(null) {}
+};
+
+template <typename Archiver>
+Archiver& operator&(Archiver& ar, const Column& s) {  // NOLINT
+    ar.StartObject();
+    ar.Member(s.name.c_str()) & hybridse::sdk::DataTypeName(s.type);
+    return ar.EndObject();
+}
+
+struct PutResp {
+    PutResp() = default;
+    int code = 0;
+    std::string msg = "ok";
+};
+
+template <typename Archiver>
+Archiver& operator&(Archiver& ar, const PutResp& s) {  // NOLINT
+    ar.StartObject();
+    ar.Member("code") & s.code;
+    ar.Member("msg") & s.msg;
+    return ar.EndObject();
+}
+
+struct ExecSPResp {
+    ExecSPResp() = default;
+    int code = 0;
+    std::string msg = "ok";
+    struct Data {
+        SimpleSchema schema;
+        std::shared_ptr<hybridse::sdk::ResultSet> rs;
+    };
+    Data data;
+};
+
+template <typename Archiver, typename Type>
+void WriteArray(Archiver& ar, const std::string& name, const std::vector<Type>& vec) {  // NOLINT
+    ar.Member(name.c_str());
+    size_t count = vec.size();
+    ar.StartArray();
+    for (size_t i = 0; i < count; i++) {
+        ar& vec[i];
+    }
+    ar.EndArray();
+}
+
+template <typename Archiver>
+void WriteValue(Archiver& ar, std::shared_ptr<hybridse::sdk::ResultSet> rs, const SimpleSchema& schema,  // NOLINT
+                int i) {
+    if (rs->IsNULL(i)) {
+        if (!schema[i].is_null) {
+            LOG(ERROR) << "Value in " << schema[i].name << " is null but it can't be null";
+        }
+        ar.SetNull();
+        return;
+    }
+    switch (schema[i].type) {
+        case hybridse::sdk::kTypeInt32: {
+            ar & rs->GetInt32Unsafe(i);
+            break;
+        }
+        case hybridse::sdk::kTypeInt64: {
+            ar & rs->GetInt64Unsafe(i);
+            break;
+        }
+        case hybridse::sdk::kTypeInt16: {
+            ar& static_cast<int>(rs->GetInt16Unsafe(i));
+            break;
+        }
+        case hybridse::sdk::kTypeFloat: {
+            ar& static_cast<double>(rs->GetFloatUnsafe(i));
+            break;
+        }
+        case hybridse::sdk::kTypeDouble: {
+            ar & rs->GetDoubleUnsafe(i);
+            break;
+        }
+        case hybridse::sdk::kTypeString: {
+            ar & rs->GetStringUnsafe(i);
+            break;
+        }
+        case hybridse::sdk::kTypeTimestamp: {
+            ar & rs->GetTimeUnsafe(i);
+            break;
+        }
+        case hybridse::sdk::kTypeDate: {
+            ar & rs->GetDateUnsafe(i);
+            break;
+        }
+        case hybridse::sdk::kTypeBool: {
+            ar & rs->GetBoolUnsafe(i);
+            break;
+        }
+        default: {
+            LOG(ERROR) << "Invalid Column Type";
+            ar & "err";
+            break;
+        }
+    }
+}
+
+template <typename Archiver>
+Archiver& operator&(Archiver& ar, const ExecSPResp& s) {  // NOLINT
```

Comment:
why overload the global &

Reply:
In this way, we can use 'writer & <struct>' to write, simpler to write.